pkg/cloud/client.go

```diff
@@ -23,27 +23,26 @@ import (
 	infrav1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
 	"github.com/pkg/errors"
 	"gopkg.in/ini.v1"
-	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 //go:generate mockgen -destination=../mocks/mock_client.go -package=mocks github.com/aws/cluster-api-provider-cloudstack/pkg/cloud Client
 
 type Client interface {
-	GetOrCreateCluster(*infrav1.CloudStackCluster) error
-	GetOrCreateVMInstance(*infrav1.CloudStackMachine, *capiv1.Machine, *infrav1.CloudStackCluster, string) error
-	ResolveVMInstanceDetails(*infrav1.CloudStackMachine) error
-	DestroyVMInstance(*infrav1.CloudStackMachine) error
-	AssignVMToLoadBalancerRule(*infrav1.CloudStackCluster, string) error
+	ClusterIface
+	VMIface
 	ResolveNetwork(*infrav1.CloudStackCluster) error
 	GetOrCreateNetwork(*infrav1.CloudStackCluster) error
 	OpenFirewallRules(*infrav1.CloudStackCluster) error
 	ResolvePublicIPDetails(*infrav1.CloudStackCluster) (*cloudstack.PublicIpAddress, error)
 	ResolveLoadBalancerRuleDetails(*infrav1.CloudStackCluster) error
 	GetOrCreateLoadBalancerRule(*infrav1.CloudStackCluster) error
+	AffinityGroupIFace
 }
 
 type client struct {
 	cs *cloudstack.CloudStackClient
+	// This is a placeholder for sending non-blocking requests.
+	// csA *cloudstack.CloudStackClient
```

Comment:
I wanted to leave this in here. When we want to send a delete command without waiting, this is how we'll do it.

Reply:
I'm not a huge fan of leaving comments to help with future work, especially minor things like syntax. I'd be inclined to remove, especially since there's no timeline for this work actually being done